### PR TITLE
🚀 Don't ship toggleAttribute innards if using an ESM build

### PR DESCRIPTION
**Pull request by kristoferbaxter**
If using ESM build, rely on native `Element.toggleAttribute()`.

### CODE REVIEW

File: src/dom.js [hunk 1]
```diff
@@ -938,6 +938,9 @@ export function domOrderComparator(element1, element2) {
  * @return {boolean} Whether or not the element now has the attribute.
  */
 export function toggleAttribute(element, name, forced) {
+  if (IS_ESM) {
+    return element.toggleAttribute(name, forced);
```
Comment (rcebulko) [inline src/dom.js:942]: WDYT about moving this to a separate polyfill so the codebase can use toggleAttribute throughout?

Comment (rcebulko) [reply src/dom.js:942]: (looks like a few files in amp-story code already use it too)

Comment (rcebulko) [reply src/dom.js:942]: Counter-point to myself: this minifies better as a single function call versus actually including `toggleAttribute` everywhere, which can't be minified

**Review by rcebulko — APPROVED** [review]

File: src/dom.js (hunk 1 above)
Comment (kristoferbaxter) [reply src/dom.js:942]: I was hoping to keep this PR smaller in scope, but yes to polyfilling only where necessary and calling the native method name in a future change.

Comment (rcebulko) [reply src/dom.js:942]: If you file the issue I'll take care of it

**Comment by kristoferbaxter** [comment]
This PR is blocked on `check-types` failing with Babel pipeline replacements for defined items like `IS_ESM`.

@rcebulko are you working on a solution here or should I exempt this file for now?

**Comment by rcebulko** [comment]
> This PR is blocked on `check-types` failing with Babel pipeline replacements for defined items like `IS_ESM`.
> 
> @rcebulko are you working on a solution here or should I exempt this file for now?

Solution is in https://github.com/ampproject/amphtml/pull/34259

File: build-system/test-configs/conformance-config.textproto [hunk 2]
```diff
@@ -17,12 +17,6 @@ requirement: {
   allowlist: 'src/layout.js'
 }
 
-requirement: {
-  type: BANNED_PROPERTY_CALL
-  error_message: 'toggleAttribute is not supported in IE and iOS prior to 13.3. Switch to setAttribute and removeAttribute.'
-  value: 'Element.prototype.toggleAttribute'
```
Comment (rcebulko) [inline build-system/test-configs/conformance-config.textproto:23]: Let's not remove this altogether until the polyfill is in place and instead just add `allowlist: 'src/dom.js'`

**Comment by kristoferbaxter** [comment]
Closing this PR since our browser support policy requires polyfilling even the module case.

**Pull request closed without merging**